### PR TITLE
Improve proompting and send typing earlier for images

### PR DESCRIPTION
**Pull request by hollowstrawberry**

### CODE REVIEW

File: ai_user/prompts/image/base.py [hunk 1]
```diff
@@ -38,8 +38,7 @@ async def _create_prompt(self, bot_prompt) -> Optional[list[dict[str, str]]]:
         return prompt
 
     async def _process_image(self, image: Image, bot_prompt: str) -> Optional[list[dict[str, str]]]:
-        raise NotImplementedError(
-            "_process_image() must be implemented in subclasses")
+        raise NotImplementedError("_process_image() must be implemented in subclasses")
```
Comment (zhaobenny) [inline ai_user/prompts/image/base.py:41]: What autoformatter are you using?
I can switch to avoid this sorta changes

Comment (hollowstrawberry) [reply ai_user/prompts/image/base.py:41]: Oh feel free to undo those. I tend to compact lines as long as they don't go past 120 characters wide

File: ai_user/prompts/base.py [hunk 2]
```diff
@@ -78,24 +78,22 @@ async def _handle_reply_in_history(self, history: list, message: Message):
...
-                {"role": "system", "content": f'The following message is a reply to: User "{replied_message.author.name}" said: {replied_message.content}'})
+                {"role": "system", "content": f'[The following message is a reply to:]\n'
+                                              f'{replied_message.author.name}: {message_content}'})
 
     def _format_message(self, message: Message) -> Dict[str, str]:
         """ Formats a message into a JSON format for OpenAI """
         role = "user" if message.author != self.bot else "assistant"
         message_content = self._mention_to_text(message)
-        content = f'User "{message.author.name}" said: {message_content}' if role == "user" else message_content
+        content = f"{message.author.name}: {message_content}"
```
[the remark this answers is not in the document]
Comment (hollowstrawberry) [reply ai_user/prompts/base.py:91]: Ah, I hadn't foreseen that. Quite a fun name you got.

I can attest that the [Image] and [Embed] are an improvement, but this is neither better or worse (just saves a few tokens). I'll undo it tomorrw or you can do it if you want to merge today.

File: ai_user/prompts/image/base.py (hunk 1 above)
Comment (hollowstrawberry) [reply ai_user/prompts/image/base.py:41]: I use PyCharm for the most part, but it doesn't reformat code for me unless I paste it in

Comment (zhaobenny) [reply ai_user/prompts/image/base.py:41]: Should be fine, I'll try to change the autoformatter settings

File: ai_user/prompts/base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (zhaobenny) [reply ai_user/prompts/base.py]: 🎉

**Pull request merged**